lib/core/utils/location_utils.dart

```diff
@@ -0,0 +1,54 @@
+import 'package:geolocator/geolocator.dart';
+
+class LocationUtils {
+  LocationPermission _permissionStatus = LocationPermission.denied;
+  final Function()? onPermissionDeniedForever;
+
+  LocationUtils({
+    this.onPermissionDeniedForever,
+  });
+
+  LocationPermission get permissionStatus => _permissionStatus;
+
+  Future<Position> getCurrentLocation() async {
+    if (!(await Geolocator.isLocationServiceEnabled())) {
+      throw LocationServiceNotEnabledException();
+    }
+    
+    if (!await _checkAndRequestPermission()) {
+      throw PermissionNotGrantedException();
+    }
+
+    return await Geolocator.getCurrentPosition();
+  }
+
+  Future<bool> _checkAndRequestPermission() async {
+    LocationPermission _status = await Geolocator.checkPermission();
```

Comment:
Aware about request Android location permission

Also need show way to redirect to app in iOS so user can re-enabled location again in iOS settings @mthinh

Reply:
Yes we do have function to navigate to app's setting using Geolocator package but I didn't specify it here, may be I will define more props callback like 
`onPermissionDenied` 
`onPermissionDeniedForever` 
I think by using this, it's easier for us if we need to modify custom behavior base on each screen logic

Reply:
Nice, if we aware about this, it's okay, let merged and build test around both platform 💯